src/SmartTransactionsController.ts

```diff
@@ -192,16 +189,18 @@ export default class SmartTransactionsController extends StaticIntervalPollingCo
     this.initializeSmartTransactionsForChainId();
     this.#ensureUniqueSmartTransactions();
 
-    onNetworkStateChange(({ providerConfig: newProvider }) => {
-      const { chainId } = newProvider;
+    onNetworkStateChange((networkState) => {
+      const { selectedNetworkClientId } = networkState;
+      const networkClient = this.getNetworkClientById(selectedNetworkClientId);
+      const { chainId } = networkClient.configuration;
       const isNewChainId = chainId !== this.config.chainId;
       this.configure({ chainId });
       this.initializeSmartTransactionsForChainId();
       if (isNewChainId) {
         this.#ensureUniqueSmartTransactions();
       }
       this.checkPoll(this.state);
-      this.ethQuery = new EthQuery(provider);
+      this.ethQuery = new EthQuery(networkClient.provider);
     });
```

Comment:
Nit: Can we make it this way instead. the only difference is less rows ?
```suggestion
 onNetworkStateChange(({ selectedNetworkClientId }) => {
    const {
      configuration: { chainId },
      provider,
    } = this.getNetworkClientById(selectedNetworkClientId);
    const isNewChainId = chainId !== this.config.chainId;
    this.configure({ chainId });
    this.initializeSmartTransactionsForChainId();
    if (isNewChainId) {
      this.#ensureUniqueSmartTransactions();
    }
    this.checkPoll(this.state);
    this.ethQuery = new EthQuery(provider);
  });
```